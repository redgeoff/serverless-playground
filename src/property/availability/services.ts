import ResponseError from '../../utils/response-error';

export interface Availability {
  foo: string;
  propertyId?: string;
}

export class BadPropertyError extends ResponseError {
  constructor(message: string) {
    super('BAD_PROPERTY', message);
  }
}

export type GetAvailability = (propertyId?: string) => Promise<Availability>;

// Note: async is not needed here, but is used as a lot of real-world examples will need async
export const availability: GetAvailability = async (propertyId?: string) => {
  if (propertyId === '123') {
    throw new BadPropertyError('cannot be 123');
  }

  return { foo: 'bar', propertyId };
};
