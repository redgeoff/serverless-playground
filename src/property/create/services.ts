export interface CreateProperty {
  id?: string;
  name: string;
}

export interface Property extends CreateProperty {
  id: string;
}

const properties: any = {};

export type CreatePropertyType = (
  property: CreateProperty
) => Promise<Property>;

// Note: async is not needed here, but is used as a lot of real-world examples will need async
export const createProperty: CreatePropertyType = async (
  property: CreateProperty
) => {
  const prop = { id: property.name, ...property };
  properties[prop.id] = prop;
  return prop;
};

// Note: async is not needed here, but is used as a lot of real-world examples will need async
export const getProperty = async (id: string) => {
  return properties[id];
};
