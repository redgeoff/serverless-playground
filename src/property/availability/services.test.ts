import { availability } from './services';

it('should get availability', async () => {
  expect(await availability('1234')).toMatchSnapshot();

  await expect(availability('123')).rejects.toEqual(new Error('cannot be 123'));
});
