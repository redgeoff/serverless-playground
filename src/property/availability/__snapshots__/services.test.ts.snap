// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get availability 1`] = `
Object {
  "foo": "bar",
  "propertyId": "1234",
}
`;
